Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
         // If url is valid, check if `handleDeepLink` method is called
         XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
     }
+
+    // MARK: - Track open
+
+    @MainActor
+    func test_onInlineMessageShown_expectTrackOpenedMetric_expectGlobalMessageShownListener() async {
+        let inlineView = await showInlineMessageForMetrics()
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
```

Comment:
I think we should move assertions dealing with the global event listener out of these test functions. 

I also consider tests for the global event listener out of scope for this ticket.

Reply:
Since some cases of [global event listeners](https://linear.app/customerio/issue/MBL-330/send-events-to-gist-event-listeners) are handled in this ticket as these both are linked to each other, and I have mentioned in the other [PR](https://github.com/customerio/customerio-ios/pull/775) that certain cases for messageShown and messageActionTaken are taken care of in this PR, I think these cases are relevant here.